src/schema/sources.ts

```diff
@@ -1121,6 +1138,30 @@ export const resolvers: IResolvers<any, Context> = {
       }
       return res[0];
     },
+    relatedTags: async (_, { sourceId }, ctx): Promise<GQLTagResults> => {
```

Comment:
@idoshamun Mostly from query point of view, any suggestions here or this will do for now?

Reply:
that looks ok! let's monitor it and if needed introduce redis in front

Reply:
We can also consider some graphQl magic on FE with high stale time and combing the queries into one request.

Reply:
For tag recommendation of onboarding we used materialized view which we refresh in cron once a week.

Reply:
We can start with this and monitor if we need more if that works for you?

Reply:
Yeah, from my side this looks ok for now.